packages/contract-schema/index.js

```diff
@@ -176,7 +176,8 @@ var properties = {
   },
   networkType: {},
   devdoc: {},
-  userdoc: {}
+  userdoc: {},
+  "db:contract": {}
```

Comment:
Why not have a new schema for `db` where `contract` is defined? Seems like that's how it's going to be implemented, so I'm not sure why we should bundle it up like this. Not picky, just food for thought

Reply:
My reasoning for adding it here is so that, if you have `truffle compile` hooked up to save to @truffle/db, then it would also include this correlation information for future use, like with a subsequent `truffle migrate`.

I'm not sure what you mean by "new schema for `db`", could you expand on that? A new schema alongside `contract-schema.spec.json`?

Reply:
> My reasoning for adding it here is so that, if you have truffle compile hooked up to save to @truffle/db, then it would also include this correlation information for future use, like with a subsequent truffle migrate.

Ya I'm cool with `db` living in this schema, but I'm mainly concerned about the use of `:` and see how there may be future `db:*` fields. So why not make it `db` which points to some other type/schema which then defines `contract`?

Reply:
I first considered a `db` field for this, containing `contract`, but my reasoning against it is that I can see it looking like this:
```javascript
{
  // ...

  db: {
   contract: { id: "0x..." }
  }

  // ...

  networks: {
    // ...

    "1": {
      // ...

      db: {
        contractInstance: { id: "0x..." }
      }
    }
  }
}
```

Which, I dunno, I guess it's fine, but it just kind of irks me? I will say: looking at it now, it irks me less than when I first considered it.

Reply:
frankly, the only thing that irks me in your example is the fact we have two things that are so close in name and imply two different things (`contract` and `contractInstance`). I'm kind of fine with the multiple usages of `db` which have different types/schemas depending on their context (i.e. `networks["1"].db` shouldn't have a `contract` and `db` should not have a `contractInstance`)

But it seems like you understand my point. I'm not picky either way here, but do please consider how it may change. I don't want to see this becoming

``` json5
{
  // ...
  "db:contract": { id: "0x..." }
  "db:<somefuturething>": { /* .... */ }
}
```

when it's quite clear the above would be sensible to implement as 

``` json5
{
  // ...
  db: {
    contract: { id: "0x..." },
    somefuturething: { /* .... */ }
  }
}
```

Reply:
The thing is: I don't think we'll add another `db:*` alongside an existing one. At the top level, it's a Contract resource. At the network level, it's a ContractInstance resource. With those two IDs you have access to everything else that would be represented by the artifact. This seems very unlikely to change.



> we have two things that are so close in name and imply two different things (`contract` and `contractInstance`).

They are indeed very different things. We had a naming discussion long ago, where the competing argument was to call it simply "Instance" rather than "ContractInstance". But clarity won out.

Reply:
(Oh, just in case it's not clear: I'm not at all trying to argue with you. I recognize all your points, for sure. I just want to provide context about my thinking so you know where I'm coming from)

Reply:
So it turned out to be better in implementation to do `db` instead of `"db:contract"`. My original thinking that "it wasn't necessary to include more than the contract" has turned into "actually we can get a bunch of other stuff in here for free that might come in handy"

Reply:
I am only seeing this discussion now but am glad for the decision to do `db` instead. I think it makes sense to include the other resources as it seems you have now!